.github/workflows/backend.yml

```diff
@@ -186,7 +168,21 @@ jobs:
           filters: |
             all:
               - added|modified: '**/*.py'
-              - added|modified: 'requirements-base.txt'
+              - added|modified: 'requirements-dev-only-frozen.txt'
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: |
+            requirements-dev-only-frozen.txt
+
+      - name: Setup pre-commit
+        # We don't use make setup-git because we're only interested in installing
+        # requirements-dev-only-frozen.txt as a fast path.
+        run: |
+          pip install -r requirements-dev-only-frozen.txt
+          pre-commit install --install-hooks
```

Comment:
we can probably just use `pre-commit install-hooks` too since we don't actually care about `.git/hooks` getting in the way

Reply:
Oh, true.